api/base/views.py

```diff
@@ -437,7 +453,93 @@ def root(request, format=None, **kwargs):
     if utils.has_admin_scope(request):
         return_val['meta']['admin'] = True
 
-    return Response(return_val)
+    resp = Response(return_val)
+
+    # This is used current for our partnership with Sloan and can be deleted after their study is complete.
+    for key, value in cookies.items():
+        resp.set_cookie(key, value)
+
+    return resp
+
+
+def sloan_study_disambiguation(request):
+    """
+    This is a hack to set flags and cookies for out Sloan study, it can be deleted when the study is complete.
+    """
+    user = request.user
+    cookies = {}
+
+    flags = []
+    sloan_data = {}
+    for flag in Flag.objects.all():
+        active = flag.is_active(request._request)
+        if flag.name in SLOAN_FLAGS and flag.everyone is None:
+            sloan_data[flag.name] = active
+
+        # User tags should override any cookie info
+        if user and not user.is_anonymous:
+            if user.all_tags.filter(name=flag.name):
+                sloan_data[flag.name] = True
+                active = True
+            if user.all_tags.filter(name=f'no_{flag.name}'):
+                sloan_data[flag.name] = False
+                active = False
+
+        if active:
+            flags.append(flag.name)
+
+    referer_url = request.environ.get('HTTP_REFERER', '')
+    provider = get_provider_from_url(referer_url)
+
+    if provider and provider.in_sloan_study:
+        for key, value in sloan_data.items():
+            cookies.update(set_tags_and_cookies_for_sloan(user, key, value))
+
+    return flags, cookies
+
+
+def get_provider_from_url(referer_url: str) -> PreprintProvider:
```

Comment:
🏅 for type hints!

this might return `None`, tho
```suggestion
def get_provider_from_url(referer_url: str) -> Optional[PreprintProvider]:
```